### PR TITLE
Support Numpy >= 2.0.0 and Pandas >= 2.0.0

### PR DESCRIPTION
**Pull request by mkopec87**
* Support numpy >= 2.0.0 (np.string_ -> np.bytes_).
* Fix test utility corner case error (test_numpy.twosigfigs function).
* Fix typo in build pipeline Python versions config list.
* aaand other stuff I'm discovering along the way ...

### CODE REVIEW

**Comment by mbaak** [comment]
Hi Mateusz, I made this PR that fixes the histogrammar unit tests for numpy<2.0.0 and pandas<2.0.0:
https://github.com/histogrammar/histogrammar-python/pull/66
They were minor issues only. I asked Simon to do a short review, or perhaps you can have a quick look. 

For your PR, once this one is merged can you please rebase? Once your PR is merged in I will switch to pyproject.toml, then release the new patch version, and then switch to upgrading popmon. 

Btw, thanks for the work you already put in, I appreciate it.

**Comment by mkopec87** [comment]
Thanks @mbaak !

Shall I include version bump / changelog / README updates in my PR or this will be part of pyproject.toml/release version PR ?

**Comment by mbaak** [comment]
I thought of adding my (small) fixes to the changelog & README after your fixes are in, so for the new release.
(But you can add it too if you want.)

**Comment by mkopec87** [comment]
Still getting one error in the tests:

```
self = <tests.test_numpy.TestPandas testMethod=test_n_bins>

    def test_n_bins(self):
        """ Test getting the number of allocated bins
        """
        with Pandas() as pd, Numpy() as np:  # noqa
            if pd is None or np is None:
                return
            sys.stderr.write("\n")
    
            df, hist1, hist2, hist3 = get_test_histograms1()
    
            assert hist1.n_bins == 5
            assert hist2.n_bins == 5
>           assert hist3.n_bins == 7
E           assert 5 == 7
E            +  where 5 = <SparselyBin binWidth=86400000000000.0 bins=Bin nanflow=Count>.n_bins

test_numpy.py:714: AssertionError
```

This test was not really testes in master because of error in Pandas() context manager.

**Comment by mkopec87** [comment]
Fixed the test as you suggested, ready for review :)

File: histogrammar/util.py [hunk 1]
```diff
@@ -318,7 +318,7 @@ def function(datum):
                         context.update(datum)                  # use its items as variables
 
                     # if the datum is a Numpy record array, override the namespace with its field names
-                    elif numpy is not None and isinstance(datum, numpy.core.records.recarray):
+                    elif numpy is not None and isinstance(datum, numpy._core.records.recarray):
```
[the remark this answers is not in the document]
Comment (mkopec87) [reply histogrammar/util.py]: Done. Would make sense to add this both setups into build test matrix though, maybe as part of solving #67 ?

[the remark this answers is not in the document]
Comment (mbaak) [reply histogrammar/util.py]: Not sure I understand, what is your suggestion?

[the remark this answers is not in the document]
Comment (mkopec87) [reply histogrammar/util.py]: To have unit tests being run in build pipeline (github actions) with two requirement setups:
- latest pandas and numpy
- pandas and numpy < 2
As I assume you want to keep the library working for both setups?

Currently there is a test matrix for different python versions, we could add also another dimension of numpy/pandas versions setting.

[the remark this answers is not in the document]
Comment (mbaak) [reply histogrammar/util.py]: Clear. Yes, can do it there.
